test/test_node.py

```diff
@@ -413,6 +413,197 @@ class DoubleParameter(cgp.OperatorNode):
     assert y == pytest.approx(p + q)
 
 
+# Node definition currently within test,
+# since definition is only in example_composed_target_function
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than 0
+     and outputs input1 if true and input2 otherwise """
+
+    _arity = 3
+
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        if inputs[0] > 0:
+            self._output = inputs[1]
+        else:
+            self._output = inputs[2]
+
+    _def_output = "x_1 if x_0 > 0 else x_2"
+    _def_numpy_output = "np.piecewise(x_0, [x_0> 0, x_0<=0], [x_1 [x_0> 0] , x_2[x_0<=0]])"
+    _def_sympy_output = "Piecewise((x_1, x_0 > 0), (x_2, x_0 <= 0))"
+    _def_torch_output = "torch.where(x_0 > 0, x_1, x_2)"
+
+
+# Node definition currently within test,
+# since definition is only in example_composed_target_function
+class ComparisonOperator(cgp.OperatorNode):
+    """A node that compares its inputs and outputs 1.0 if the first is bigger
+     and 0 otherwise """
+
+    _arity = 2
+    _def_output = "max(1.0 * ((x_0 - x_1)/max(abs(x_0 - x_1), 0.000001)), 0.0)"
+    _def_sympy_output = "Max(1.0 * sign(x_0 - x_1), 0.0)"
+    _def_torch_output = "torch.max(1.0 * torch.sign(x_0 - x_1), torch.zeros(x_0.shape))"
+    _def_numpy_output = "np.maximum(1.0 * np.sign(x_0 - x_1), np.zeros(x_0.shape))"
+
+
+def test_comparison_operator():
+
+    genome_params = {
+        "n_inputs": 2,
+        "n_outputs": 1,
+        "n_columns": 1,
+        "n_rows": 1,
+        "levels_back": None,
+    }
+    primitives = (ComparisonOperator,)
+    genome = cgp.Genome(**genome_params, primitives=primitives)
+    genome.dna = [
+        ID_INPUT_NODE,
+        ID_NON_CODING_GENE,
+        ID_NON_CODING_GENE,
+        ID_INPUT_NODE,
+        ID_NON_CODING_GENE,
+        ID_NON_CODING_GENE,
+        0,  # function gene
+        0,  # first input gene is address of first input node
+        1,  # second input gene is address of second input node
+        ID_OUTPUT_NODE,
+        2,
+        ID_NON_CODING_GENE,
+    ]
+
+    x_0 = [1.0, 0.0]
+    y_target_0 = [1.0]
+    _test_graph_call_and_to_x_compilations(genome, x_0, y_target_0)
+
+    x_1 = [1.0, 1.0]
+    y_target_1 = [0.0]
+    _test_graph_call_and_to_x_compilations(genome, x_1, y_target_1)
+
+    x_2 = [0.0, 1.0]
+    y_target_2 = [0.0]
+    _test_graph_call_and_to_x_compilations(genome, x_2, y_target_2)
+
+
+def test_if_else_operator():
+
+    genome_params = {
+        "n_inputs": 3,
+        "n_outputs": 1,
+        "n_columns": 1,
+        "n_rows": 1,
+        "levels_back": None,
+    }
+    primitives = (IfElseOperator,)
+    genome = cgp.Genome(**genome_params, primitives=primitives)
+    genome.dna = [
```

Comment:
please add a comment indicating which function this genome encode like `# f(x) = ...`